scripts/lib/api/objectsInv.ts

```diff
@@ -0,0 +1,170 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { readFile, writeFile } from "fs/promises";
+import { unzipSync, deflateSync } from "zlib";
+import { removeSuffix } from "../stringUtils";
+import { join, dirname } from "path";
+import { mkdirp } from "mkdirp";
+
+/**
+ * Some pages exist in the sphinx docs but not in our docs
+ * If any URIs match these cases, we remove their entries.
+ **/
+const ENTRIES_TO_EXCLUDE = [
+  /^genindex(\.html)?$/,
+  /^py-modindex(\.html)?$/,
+  /^search(\.html)?$/,
+  /^explanation(\.html)?(?=\/|#|$)/,
+  /^how_to(\.html)?(?=\/|#|$)/,
+  /^tutorials(\.html)?(?=\/|#|$)/,
+  /^migration_guides(\.html)?(?=\/|#|$)/,
+  /^configuration(\.html)?(?=#|$)/,
+  /^contributing_to_qiskit(\.html)?(?=#|$)/,
+  /^deprecation_policy(\.html)?(?=#|$)/,
+  /^faq(\.html)?(?=#|$)/,
+  /^getting_started(\.html)?(?=#|$)/,
+  /^intro_tutorial1(\.html)?(?=#|$)/,
+  /^maintainers_guide(\.html)?(?=#|$)/,
+  /^qc_intro(\.html)?(?=#|$)/,
+  /^release[-_]notes(\.html)?(?=#|$)/,
+  /^legacy_release_notes(\.html)?(?=#|$)/,
+];
+
+function shouldExcludePage(uri: string): boolean {
+  return ENTRIES_TO_EXCLUDE.some((condition) => uri.match(condition));
+}
+
+export type ObjectsInvEntry = {
+  name: string;
+  domainAndRole: string;
+  priority: string;
+  uri: string;
+  dispname: string;
+};
+
+/**
+ * Class to hold and operate on data from Sphinx's objects.inv file.
+ * For information on the syntax, see:
+ *   https://sphobjinv.readthedocs.io/en/stable/syntax.html
+ */
+export class ObjectsInv {
+  preamble: string;
+  entries: ObjectsInvEntry[];
+  constructor(preamble: string, entries: ObjectsInvEntry[]) {
```

Comment:
```suggestion
  readonly preamble: string;
  readonly entries: ObjectsInvEntry[];
  
  constructor(preamble: string, entries: ObjectsInvEntry[]) {
```

Reply:
```suggestion
  readonly preamble: string;
  entries: ObjectsInvEntry[];
  constructor(preamble: string, entries: ObjectsInvEntry[]) {
```

Thanks! Good call for `preamble`, but `entries` is mutated in `updateUris`.